### PR TITLE
GODRIVER-1911 Pass PYTHON variable to set-temp-creds.sh

### PR DESCRIPTION
**Pull request by divjotarora**

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -304,6 +303,15 @@ functions:
...
             ${PYTHON3_BINARY} -m venv ./venv
+
+            # Set the PYTHON environment variable to point to the active python3 binary. This is used by the
+            # set-temp-creds.sh script.
+            if [ "Windows_NT" = "$OS" ]; then
+              export PYTHON="$(pwd)/venv/Scripts/python"
+            else
+              export PYTHON="$(pwd)/venv/bin/python"
+            fi
+
```
Comment (benjirewis) [inline .evergreen/config.yml:314]: Ideally, we'd combine the virtual environment creation, python3 binary setting, and boto3 installation into a single call to [activate_venv.sh.](https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/auth_aws/activate_venv.sh). But, that unfortunately sets `PYTHON_BINARY` and not `PYTHON`, so it wouldn't work perfectly...

File: .evergreen/config.yml [hunk 2]
```diff
@@ -75,8 +75,7 @@ functions:
           if [ "Windows_NT" = "$OS" ]; then
              mkdir -p c:/libmongocrypt/include
              mkdir -p c:/libmongocrypt/bin
-             # TODO: After a stable libmongocrypt 1.1.0 is released in MONGOCRYPT-293, update this URL to: https://s3.amazonaws.com/mciuploads/libmongocrypt/windows/latest_release/libmongocrypt.tar.gz
-             curl https://s3.amazonaws.com/mciuploads/libmongocrypt/windows/latest_release/libmongocrypt_unstable.tar.gz --output libmongocrypt.tar.gz
+             curl https://s3.amazonaws.com/mciuploads/libmongocrypt/windows/latest_release/libmongocrypt.tar.gz --output libmongocrypt.tar.gz
```
Comment (benjirewis) [inline .evergreen/config.yml:78]: Nice, looks like there was a TODO all along.

**Review by benjirewis — APPROVED** [review]
LGTM. Thank you for investigating/fixing!

File: .evergreen/config.yml (hunk 1 above)
Comment (divjotarora) [reply .evergreen/config.yml:314]: Also, `PYTHON3_BINARY` points to the system installation of python3, but we actually want to point to the virtualenv python installation here.

Comment (benjirewis) [reply .evergreen/config.yml:314]: Ah yep, that too.

**Review by iwysiu — APPROVED** [review]

**Pull request merged**